### ISSUE TITLE
AccessViolationException while converting video frames to bmp to png in v4.1.1.3497

### ISSUE THREAD

**Issue by smzinovyev** [emgucv/emgucv#290]
I am using the VideoSurveillance sample project. As I want to see frames, I convert each frame.Bitmap to png and save to disk. It was not the main goal of my home project but while working on it I have got AccessViolationException. I tried to minimize number of lines to get the root cause. And I thought it will be good to report it as an issue.

My project is a Windows Console app targeting .Net4.8. I am using VS2019 on Windows 10 Pro.
It references EMGU.CV version="4.1.1.3497"

```
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Drawing.Imaging;
using System.IO;
using System.Linq;
using Emgu.CV;
using Emgu.CV.Structure;
using Emgu.CV.VideoStab;

namespace Motion
{
    public static class Program
    {
        public static void Main(string[] args)
        {
            string filePath = "Megamind.avi";
            FileInfo fileInfo = new FileInfo(filePath);
            if (!fileInfo.Exists)
            {
                throw new InvalidOperationException($"File not found '{filePath}'.");
            }

            string fileName = Path.GetFileNameWithoutExtension(filePath);

            DirectoryInfo dirPng = Directory.CreateDirectory(Path.Combine(fileInfo.DirectoryName, "!PNG"));
            Directory.Delete(dirPng.FullName, true);
            Directory.CreateDirectory(dirPng.FullName);


            List<Bitmap> bitmaps = new List<Bitmap>();

            using (VideoCapture vc = new VideoCapture(fileInfo.FullName))
            using (CaptureFrameSource cfs = new CaptureFrameSource(vc))
            {
                int i = 0;
                Mat f;
                while ((f = cfs.NextFrame()) != null)
                {
                    if (f.Width <= 0)
                    {
                        break; // while
                    }

                    // 0 and 1 are enough to reproduce AccessViolationException
                    if (i > 1)
                    {
                        break; // while
                    }

                    // **************************************************************
                    // SZ 2020-01-29. Uncomment to reproduce AccessViolationException
                    var unused = new Image<Gray, byte>(new Size(1920, 1088));
                    // **************************************************************


                    // There is no Mat.Bitmap property in v4.2.0.3636. It was in v4.1.1.3497.
                    bitmaps.Add(f.Bitmap);

                    ++i;
                }
            }

            // Save png files
            int _ = bitmaps
                //.AsParallel()
                .Select((v, i) =>
                {
                    try
                    {
                        v?.Save(Path.Combine(dirPng.FullName, $"{fileName}_{i:D5}.png"), ImageFormat.Png);
                    }
                    catch (Exception ex)
                    {
                        Console.WriteLine(ex.ToString());
                    }
                    return 1;
                })
                .Count();
        }
    }
}

```

I am using https://github.com/opencv/opencv/blob/master/samples/data/Megamind.avi

--
Sergiy

**Comment by emgucv** [comment]
The Mat.Bitmap property in v4.1.1 return a Bitmap object that shares memory with the Mat object. As described in bold in the documentation:
http://www.emgu.com/wiki/files/4.1.1/document/html/28771a88-bafc-6ed9-836b-1dd35aca3602.htm 
If you do not intend to keep reference of the parent Mat, you should call ".ToBitmap()" function instead. "Mat.ToBitmap()" function is also available as an extension method in version 4.2.0 after you reference Emgu.CV.UI.
Using the Bitmap created from the Mat.Bitmap function, without keeping a reference in the parent Mat object, may results the the parent Mat object collected by the Garbage collector. Any access to the Bitmap therefore will attempt to access the memory of the garbage collected Mat object and will results in access violation.

**Comment by smzinovyev** [comment]
Thanks. It explains.
